renku/domain_model/dataset.py

```diff
@@ -420,6 +422,13 @@ def keywords_csv(self):
         """Comma-separated list of keywords associated with dataset."""
         return ", ".join(self.keywords)
 
+    def get_datadir(self, client) -> Path:
```

Comment:
Should we use injection here? Or maybe use `get_client()`? It's not very clear to me where we should use it and I believe we don't do it consistently.